src/server/scripts/Northrend/Ulduar/Ulduar/boss_hodir.cpp

```diff
@@ -11,6 +11,14 @@
 #include "SpellScript.h"
 #include "ulduar.h"
 
+/* This type of enums are used along all the bosses for all type of things.
+ * Using enums for integers produces warning messages due to unscoped types.
```

Comment:
enum class should solve it afaik

Reply:
Enum class will change the type of spells from int to the name of the class. This means that you can't use cast spells with an enum class since it can't be converted to integer.

Reply:
@Kitzunu 👀 on this pls.

Reply:
What about https://docs.microsoft.com/en-us/cpp/cpp/enumerations-cpp?view=msvc-160
```cpp
enum A : int; // non-scoped enum must have type specified
```

Reply:
The warning is still due to the enum being unscopped if it was a type problem we could use a static_cast<>

Reply:
I have an idea but it would take a bunch of work:
- Create a class to store spell data such as the uint64 id;
- use the enums for ease of reading but instead of passing the enum to the function use a getter from the spellclass in order to retrieve the id as an uint64;

Reply:
imho just do it as it currently is.

This is a greater issue that targets the whole project, and it is not game-breaking either.

Reply:
Agreed!